### PR TITLE
cli for machine list & get api

### PR DESCRIPTION
**Pull request by sushmak295**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/_format.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def parser(entry):
 
 
 def aks_addon_show_table_format(result):
-    def parser(entry):
+    def parser(entry):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_format.py:34]: remove the extra trailing space

File: src/aks-preview/azext_aks_preview/_format.py [hunk 2]
```diff
@@ -45,6 +45,21 @@ def parser(entry):
         return parsed.search(entry, Options(dict_cls=OrderedDict))
     return parser(result)
 
+def aks_machine_list_table_format(results):
+    return [aks_machine_show_table_format(r) for r in results]
+
+def aks_machine_show_table_format(result):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_format.py:51]: could you please add some ut for this?

File: src/aks-preview/azext_aks_preview/_params.py [hunk 3]
```diff
@@ -674,6 +674,13 @@ def load_arguments(self, _):
                    "--ignore-pod-disruption-budget", "-i"], action=get_three_state_flag(), is_preview=True,
                    help='delete an AKS nodepool by ignoring PodDisruptionBudget setting')
 
+    with self.argument_context('aks machine') as c:
+        c.argumemt('cluster_name', options_list=['--cluster-name'], help='The cluster name.')
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:678]: could omit the `options_list` if the option name is the same as parameter name (cli framework would help replace `_` with `-` and add leading `--`)

File: src/aks-preview/azext_aks_preview/_params.py [hunk 4]
```diff
@@ -674,6 +674,13 @@ def load_arguments(self, _):
...
+        c.argument('nodepool_name', options_list=['--nodepool-name'], validator=validate_nodepool_name, help='The node pool name.')
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:679]: could omit the `options_list`

File: src/aks-preview/azext_aks_preview/_params.py [hunk 5]
```diff
@@ -674,6 +674,13 @@ def load_arguments(self, _):
...
+    
+    with self.argument_context('aks machine show') as c:
+        c.argumemt('machine_name',options_list=['--machine-name'], help='to display specific information for all machines.')
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:682]: could omit the `options_list`

Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:682]: typo `argumemt` -> `argument`

File: src/aks-preview/azext_aks_preview/custom.py [hunk 6]
```diff
@@ -1425,6 +1426,12 @@ def aks_operation_abort(cmd,   # pylint: disable=unused-argument
     return sdk_no_wait(no_wait, client.begin_abort_latest_operation, resource_group_name, name, headers=headers)
 
 
+def aks_machine_list(cmd, client, resource_group_name, cluster_name, agentpool_name):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:1429]: Could you please add a live test case for these 2 new commands? Could find samples from `test_aks_commands.py`

File: src/aks-preview/HISTORY.rst [hunk 7]
```diff
@@ -12,6 +12,11 @@ To release a new version, please select a new version number (usually plus 1 to
 Pending
 +++++++
 
+0.5.157
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:15]: Please also bump  `VERSION` defined in `setup.py`

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 8]
```diff
@@ -1416,6 +1418,61 @@ def test_aks_nodepool_abort(self, resource_group, resource_group_location):
             self.check('powerState.code', 'Running')
         ])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_machine_cmds(self, resource_group, resource_group_location):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:1423]: Queued [live test pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=79878517&view=results) to validate the change.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 9]
```diff
@@ -1416,6 +1418,61 @@ def test_aks_nodepool_abort(self, resource_group, resource_group_location):
...
+        self.cmd('aks nodepool add ' \
+                 ' --resource-group={resource_group}' \
+                 ' --cluster-name={name}' \
+                 ' --name={node_pool_name} --node-count=2', checks=[
+            self.check('provisioningState', 'Succeeded')
+        ])
+
+        list_cmd = 'aks machine list ' \
+                   ' --resource-group={resource_group}' \
+                   ' --cluster-name={name} --nodepool-name={node_pool_name} -o json'
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:1456]: maybe you could test the output format in table in one of show and list

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 10]
```diff
@@ -1416,6 +1418,61 @@ def test_aks_nodepool_abort(self, resource_group, resource_group_location):
...
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_machine_cmds(self, resource_group, resource_group_location):
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'ssh_key_value': self.generate_ssh_keys()
+        })
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --enable-managed-identity' \
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:1431]: add trailing space to distinct different options
```suggestion
        create_cmd = 'aks create --resource-group={resource_group} --name={name} --enable-managed-identity ' \
```

**Pull request closed without merging**